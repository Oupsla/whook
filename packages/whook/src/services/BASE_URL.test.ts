import initBaseURL from './BASE_URL';

describe('initBaseURL', () => {
  const log = jest.fn();

  beforeEach(() => {
    log.mockReset();
  });

  it('should work with all dependencies', async () => {
    const BASE_URL = await initBaseURL({
      ENV: {},
      CONFIG: {
        name: 'project',
      },
      PROTOCOL: 'https',
      HOST: 'localhost',
      PORT: 1337,
      log,
    });

    expect({
      BASE_URL,
      logCalls: log.mock.calls.filter(args => 'stack' !== args[0]),
    }).toMatchSnapshot();
  });

  it('should work with required dependencies only', async () => {
    const BASE_URL = await initBaseURL({
      ENV: {},
      CONFIG: {
        name: 'project',
      },
      HOST: 'localhost',
      PORT: 1337,
      log,
    });

    expect({
      BASE_URL,
      logCalls: log.mock.calls.filter(args => 'stack' !== args[0]),
    }).toMatchSnapshot();
  });

  it('should work with a base URL in config', async () => {
    const BASE_URL = await initBaseURL({
      ENV: {},
      CONFIG: {
        name: 'project',
        baseURL: 'https://example.com',
      },
      HOST: 'localhost',
      PORT: 1337,
      log,
    });

    expect({
      BASE_URL,
      logCalls: log.mock.calls.filter(args => 'stack' !== args[0]),
    }).toMatchSnapshot();
  });

  it('should work with a base URL in config but in development mode', async () => {
    const BASE_URL = await initBaseURL({
      ENV: {
        DEV_MODE: '1',
      },
      CONFIG: {
        name: 'project',
        baseURL: 'https://example.com',
      },
      HOST: 'localhost',
      PORT: 1337,
      log,
    });

    expect({
      BASE_URL,
      logCalls: log.mock.calls.filter(args => 'stack' !== args[0]),
    }).toMatchSnapshot();
  });
});
