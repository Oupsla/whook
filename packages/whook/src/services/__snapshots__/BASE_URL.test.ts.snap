// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initBaseURL should work with a base URL in config 1`] = `
Object {
  "BASE_URL": "https://example.com",
  "logCalls": Array [
    Array [
      "debug",
      "游개 - Generated the BASE_URL constant:",
      "https://example.com",
    ],
  ],
}
`;

exports[`initBaseURL should work with a base URL in config but in development mode 1`] = `
Object {
  "BASE_URL": "http://localhost:1337",
  "logCalls": Array [
    Array [
      "debug",
      "游개 - Generated the BASE_URL constant:",
      "http://localhost:1337",
    ],
  ],
}
`;

exports[`initBaseURL should work with all dependencies 1`] = `
Object {
  "BASE_URL": "https://localhost:1337",
  "logCalls": Array [
    Array [
      "debug",
      "游개 - Generated the BASE_URL constant:",
      "https://localhost:1337",
    ],
  ],
}
`;

exports[`initBaseURL should work with required dependencies only 1`] = `
Object {
  "BASE_URL": "http://localhost:1337",
  "logCalls": Array [
    Array [
      "debug",
      "游개 - Generated the BASE_URL constant:",
      "http://localhost:1337",
    ],
  ],
}
`;
